Extensions/Video/videoruntimeobject.js

```diff
@@ -0,0 +1,241 @@
+ * @extends RuntimeObject
+ */
+gdjs.VideoRuntimeObject = function(runtimeScene, objectData) {
+  // Always call the base gdjs.RuntimeObject constructor.
+  gdjs.RuntimeObject.call(this, runtimeScene, objectData);
+
+  // Load any required data from the object properties.
+  this._property1 = objectData.content.property1;
+
+  // Create the renderer (see videoruntimeobject-pixi-renderer.js)
+  if (this._renderer)
+    gdjs.VideoRuntimeObjectRenderer.call(this._renderer, this, runtimeScene);
+  else this._renderer = new gdjs.VideoRuntimeObjectRenderer(this, runtimeScene);
+};
+
+gdjs.VideoRuntimeObject.prototype = Object.create(gdjs.RuntimeObject.prototype);
+gdjs.VideoRuntimeObject.thisIsARuntimeObjectConstructor = "Video::VideoObject"; //Replace by your extension + object name.
+
+gdjs.VideoRuntimeObject.prototype.getRendererObject = function() {
+  return this._renderer.getRendererObject();
+};
+
+/**
+ * Called once during the game loop, before events and rendering.
+ * @param {gdjs.RuntimeScene} runtimeScene The gdjs.RuntimeScene the object belongs to.
+ */
+gdjs.VideoRuntimeObject.prototype.update = function(runtimeScene) {
+  // This is an example: typically you want to make sure the renderer
+  // is up to date with the object.
+  this._renderer.ensureUpToDate();
+};
+
+/**
+ * Initialize the extra parameters that could be set for an instance.
+ * @private
+ */
+gdjs.VideoRuntimeObject.prototype.extraInitializationFromInitialInstance = function(
+  initialInstanceData
+) {
+  if (initialInstanceData.customSize) {
+    this.setWidth(initialInstanceData.width);
+    this.setHeight(initialInstanceData.height);
+  }
+};
+
+/**
+ * Update the object position.
+ * @private
+ */
+gdjs.VideoRuntimeObject.prototype._updatePosition = function() {
+  // This is an example: typically you want to tell the renderer to update
+  // the position of the object.
+  this._renderer.updatePosition();
+};
+
+/**
+ * Set object position on X axis.
+ */
+gdjs.VideoRuntimeObject.prototype.setX = function(x) {
+  gdjs.RuntimeObject.prototype.setX.call(this, x); // Always call the parent method first.
+  this._updatePosition();
+};
+
+/**
+ * Set object position on Y axis.
+ */
+gdjs.VideoRuntimeObject.prototype.setY = function(y) {
+  gdjs.RuntimeObject.prototype.setY.call(this, y); // Always call the parent method first.
+  this._updatePosition();
+};
+
+/**
+ * Set the angle of the object.
+ * @param {number} angle The new angle of the object
+ */
+gdjs.VideoRuntimeObject.prototype.setAngle = function(angle) {
+  gdjs.RuntimeObject.prototype.setAngle.call(this, angle); // Always call the parent method first.
+  this._renderer.updateAngle(); // Tell the renderer to update the rendered object
+};
+
+/**
+ * Set object opacity.
+ */
+gdjs.VideoRuntimeObject.prototype.setOpacity = function(opacity) {
+  if (opacity < 0) opacity = 0;
+  if (opacity > 255) opacity = 255;
+  var new_opacity = opacity / 255;
+  //this.opacity = opacity || this.getOpacity() ;
```

Comment:
Prefer having all the "state" of the object inside the gdjs.VideoRuntimeObject

Then, the job of the renderer is to read things from the object, and update the PIXI object so that it reflects the "state".

So here:

````
  if (opacity < 0) opacity = 0;
  if (opacity > 255) opacity = 255;
  this.opacity = opacity / 255;

  this._renderer.updateOpacity();
```

Renderer "just" needs to know when anything change in the state, so that it can update the PIXI object.

Reply:
You might also notice that renderers "break" encapsulation by reading value directly in their objects, rather than using getXXX (`this._object.opacity` rather than `this._object.getOpacity()`). This is because renderers are tighlty coupled to their object, so it's exceptionally ok for them to read things that would be "private" to objects.